### PR TITLE
Updates to POP, AppManager, and tests

### PR DESCRIPTION
**Pull request by MatisseHack**
* Updated POP
  - Updated `Trait` to `PlatformQuery`
  - BasePage now has `waitForPageToLeave`

* Updated `AppManager`
  - Moved `Platform` enum to it's own file
  - Removed some unused capabilities from `startApp`
  - `startApp` now uses environment variables set by Maven profiles to chose which platform to run on (matches XTC's environment variables)
  - Added `waitDriver`
  - Removed a bunch of `throws Exception` lines

### CODE REVIEW

File: src/test/java/pages/LoginPage.java [hunk 1]
```diff
@@ -1,47 +1,53 @@
...
-        screenshot("Entered username");
-        getDriver().hideKeyboard();
+    @Override
+    public PlatformQuery trait() {
+        return new PlatformQuery()
+                .setAndroid(By.id("SignInButton"))
+                .setiOS(By.id("SignInButton"));
+    }
+
+    public LoginPage EnterCredentials(String username, String password) {
```
Comment (ethanis) [inline src/test/java/pages/LoginPage.java]: Method should be `enterCredentials` to follow java style guides

File: src/test/java/pages/LoginPage.java [hunk 2]
```diff
@@ -1,47 +1,53 @@
...
-        getDriver().findElement(passwordField).sendKeys(password);
-        screenshot("Entered password");
-        getDriver().hideKeyboard();
+        driver.findElement(passwordField).click();
+        driver.findElement(passwordField).sendKeys(password);
+        label("Entered password");
+        driver.hideKeyboard();
 
         return this;
     }
```
Comment (ethanis) [inline src/test/java/pages/LoginPage.java:43]: `tapSignInButton`

File: src/test/java/pages/LoginPage.java (hunk 1 above)
Comment (MatisseHack) [reply src/test/java/pages/LoginPage.java]: Nice catch

File: src/test/java/pages/LoginPage.java (hunk 2 above)
Comment (MatisseHack) [reply src/test/java/pages/LoginPage.java:43]: Done!

**Pull request merged**